gammapy/spectrum/observation.py

```diff
@@ -388,28 +386,34 @@ def stack(cls, obs_list, group_id=None):
             backscal_off += backscal_off_data * o.off_vector.data
 
             # Exposure weighted IRFs
-            aefft_current = o.aeff.data * o.livetime
+            aeff_data = o.aeff.evaluate(fill_nan=True)
+            aefft_current = aeff_data * o.livetime
             aefft += aefft_current
             edisp_data = o.edisp.pdf_in_safe_range(o.lo_threshold, o.hi_threshold)
             aefftedisp += edisp_data.transpose() * aefft_current
-        
-        stacked_backscal_on = np.nan_to_num(backscal_on / stacked_off_counts)
-        stacked_backscal_off = np.nan_to_num(backscal_off / stacked_off_counts)
+
+        stacked_backscal_on = backscal_on / stacked_off_counts
+        stacked_backscal_off = backscal_off / stacked_off_counts
+
+        # there should be no nan values in backscal_on or backscal_off
+        idx = np.where(stacked_off_counts == 0)[0]
+        stacked_backscal_on[idx] = -1
+        stacked_backscal_off[idx] = -1
```

Comment:
Maybe make this magic fill value a class-level private constant?

These things have a tendency of getting different values in different places and causing bugs, and introducing constants with a one-line comment sometimes helps a bit maintain the code and debug issues.